src/registry_testing.jl

```diff
@@ -218,11 +218,16 @@ function test(path=pwd(); registry_deps::Vector{<:AbstractString}=String[])
                     end
                     # Test that the way Pkg loads this data works
                     Test.@test load_compat(compatfile, vnums)
-                    # Make sure the content roundtrips through decompression/compression
+                    # Make sure the content roundtrips through decompression/compression.
+                    # However, before we check for equality, we change the compat ranges
+                    # from `String`s to `VersionRanges`.
                     compressed = RegistryTools.Compress.compress(
                         compatfile, RegistryTools.Compress.load(compatfile)
                     )
-                    Test.@test compressed == compat
+                    mapdict = (f, dict) -> Dict(f(k, v) for (k, v) in dict)
```

Comment:
There's no need, nor intention, to ever transform the keys, so
```
(f, dict) -> Dict(k => f(v) for (k, v) in dict)
```
seems more natural to me. (And corresponding changes of the mapping functions.)